CMakeLists.txt

```diff
@@ -1,53 +1,51 @@
-# http://ros.org/doc/groovy/api/catkin/html/user_guide/supposed.html
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 project(depthimage_to_laserscan)
+if(NOT WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -Wall -Wextra -Wpedantic")
+endif()
 
-# Load catkin and all dependencies required for this package
-find_package(catkin REQUIRED dynamic_reconfigure image_geometry image_transport nodelet roscpp sensor_msgs)
-#find_package(OpenCV REQUIRED)
+find_package(ament_cmake REQUIRED)
 
-# Dynamic reconfigure support
-generate_dynamic_reconfigure_options(cfg/Depth.cfg)
+find_package(image_geometry REQUIRED)
+find_package(OpenCV REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES DepthImageToLaserScan DepthImageToLaserScanROS DepthImageToLaserScanNodelet
-  CATKIN_DEPENDS dynamic_reconfigure image_geometry image_transport nodelet roscpp sensor_msgs
+include_directories(include
+  ${OpenCV_INCLUDE_DIRS}
 )
 
-include_directories(include ${catkin_INCLUDE_DIRS})
+add_library(DepthImageToLaserScan
+  src/DepthImageToLaserScan.cpp
+)
+ament_target_dependencies(DepthImageToLaserScan
+  "image_geometry"
+  "OpenCV"
+  "sensor_msgs"
+)
 
-add_library(DepthImageToLaserScan src/DepthImageToLaserScan.cpp)
-target_link_libraries(DepthImageToLaserScan ${catkin_LIBRARIES})
+add_library(DepthImageToLaserScanROS
+  src/DepthImageToLaserScanROS.cpp
+)
+ament_target_dependencies(DepthImageToLaserScanROS
+  "rclcpp"
+)
+target_link_libraries(DepthImageToLaserScanROS DepthImageToLaserScan)
 
-add_library(DepthImageToLaserScanROS src/DepthImageToLaserScanROS.cpp)
-add_dependencies(DepthImageToLaserScanROS ${PROJECT_NAME}_gencfg)
-target_link_libraries(DepthImageToLaserScanROS DepthImageToLaserScan ${catkin_LIBRARIES})
+add_executable(depthimage_to_laserscan_node
+  src/depthimage_to_laserscan.cpp
+)
+target_link_libraries(depthimage_to_laserscan_node DepthImageToLaserScan DepthImageToLaserScanROS)
 
-add_library(DepthImageToLaserScanNodelet src/DepthImageToLaserScanNodelet.cpp)
-target_link_libraries(DepthImageToLaserScanNodelet DepthImageToLaserScanROS ${catkin_LIBRARIES})
+install(TARGETS depthimage_to_laserscan_node
+  DESTINATION ${CMAKE_INSTALL_PREFIX}/bin/
+)
```

Comment:
Should the headers be installed as well? Should the libraries be installed in lib?